tests/genai/judges/adapters/test_databricks_adapter.py

```diff
@@ -709,3 +711,204 @@ def test_call_chat_completions_with_use_case_not_supported(mock_databricks_rag_e
+                        ]
+                    }
+                )
+            }
+        ),
+    ]
+
+    messages = [litellm.Message(role="user", content="Extract outputs")]
+
+    def callback(content):
+        return {"result": content}
+
+    with (
+        mock.patch(
+            "mlflow.genai.judges.adapters.databricks_managed_judge_adapter.call_chat_completions",
+            side_effect=mock_responses,
+        ) as mock_call,
+        mock.patch(
+            "mlflow.genai.judges.adapters.databricks_managed_judge_adapter._process_tool_calls"
+        ) as mock_process,
+    ):
+        mock_process.return_value = [
+            litellm.Message(
+                role="tool",
+                content='{"name": "root_span", "inputs": null}',
+                tool_call_id="call_123",
+                name="get_root_span",
+            )
+        ]
+
+        result = _run_databricks_agentic_loop(
+            messages=messages,
+            trace=mock_trace,
+            on_final_answer=callback,
+        )
+
+        # Verify we looped twice
+        assert mock_call.call_count == 2
+
+        # Verify tool calls were processed
+        mock_process.assert_called_once()
+
+        # Verify final result
+        assert result == {"result": '{"outputs": "The answer is 42"}'}
+
+
+def test_agentic_loop_max_iteration_limit(mock_trace):
+    # Always return tool calls (never a final answer)
+    mock_response = AttrDict(
+        {
+            "output_json": json.dumps(
+                {
+                    "choices": [
+                        {
+                            "message": {
+                                "role": "assistant",
+                                "content": None,
+                                "tool_calls": [
+                                    {
+                                        "id": "call_123",
+                                        "type": "function",
+                                        "function": {
+                                            "name": "get_root_span",
+                                            "arguments": "{}",
+                                        },
+                                    }
+                                ],
+                            }
+                        }
+                    ]
+                }
+            )
+        }
+    )
+
+    messages = [litellm.Message(role="user", content="Extract outputs")]
+
+    def callback(content):
+        return content
+
+    with (
+        mock.patch(
+            "mlflow.genai.judges.adapters.databricks_managed_judge_adapter.call_chat_completions",
+            return_value=mock_response,
+        ) as mock_call,
+        mock.patch(
+            "mlflow.genai.judges.adapters.databricks_managed_judge_adapter._process_tool_calls"
+        ) as mock_process,
+        mock.patch(
+            "mlflow.genai.judges.adapters.databricks_managed_judge_adapter.MLFLOW_JUDGE_MAX_ITERATIONS"
```

Comment:
can we use `monkeypatch.setenv("MLFLOW_JUDGE_MAX_ITERATIONS"...)` - I think that's the general pattern for setting env vars